skyportal/handlers/api/internal/recent_sources.py

```diff
@@ -15,83 +15,82 @@
-            source_entry = (
-                Source.query_records_accessible_by(self.current_user)
-                .filter(Source.obj_id == obj_id)
-                .order_by(desc(Source.created_at))
-                .offset(recency_index)
-                .first()
-            )
+                s = session.scalars(
+                    Obj.select(
+                        session.user_or_token, options=[joinedload(Obj.thumbnails)]
+                    ).where(Obj.id == obj_id)
+                ).first()
 
-            sources.append(
-                {
-                    'obj_id': s.id,
-                    'ra': s.ra,
-                    'dec': s.dec,
-                    'created_at': source_entry.created_at,
-                    'thumbnails': [
-                        {
-                            "type": t.type,
-                            "is_grayscale": t.is_grayscale,
-                            "public_url": t.public_url,
-                        }
-                        for t in sorted(s.thumbnails, key=lambda t: t_index(t.type))
-                    ],
-                    'classifications': s.classifications,
-                    'recency_index': recency_index,
-                }
-            )
+                # Get the entry in the Source table to get the accurate saved_at time
+                source_entry = session.scalars(
+                    Source.select(session.user_or_token)
+                    .where(Source.obj_id == obj_id)
+                    .order_by(desc(Source.created_at))
+                    .offset(recency_index)
+                ).first()
 
-        for source in sources:
-            num_times_seen = sources_seen[source["obj_id"]]
-            # If this source was saved multiple times recently, and this is not
-            # the oldest instance of an object being saved (highest recency_index)
-            if num_times_seen > 1 and source["recency_index"] != num_times_seen - 1:
-                source["resaved"] = True
-            else:
-                source["resaved"] = False
-            # Delete bookkeeping recency_index key
-            del source["recency_index"]
+                sources.append(
+                    {
+                        'obj_id': s.id,
+                        'ra': s.ra,
+                        'dec': s.dec,
+                        'created_at': source_entry.created_at,
+                        'thumbnails': [
+                            {
+                                "type": t.type,
+                                "is_grayscale": t.is_grayscale,
+                                "public_url": t.public_url,
+                            }
+                            for t in sorted(s.thumbnails, key=lambda t: t_index(t.type))
+                        ],
+                        'classifications': s.classifications,
+                        'recency_index': recency_index,
+                    }
+                )
 
-        self.verify_and_commit()
```

Comment:
Are you sure this works without committing?

Reply:
It's just a get.